detection-rules/link_contains_punycode_characters.yml

```diff
@@ -0,0 +1,15 @@
+name: "Link to a Domain with Punycode Characters"
+description: |
+  The body contains a link to a domain with Punycode characters to hide the true URL destination, or contains non-printable ASCII content.
+references:
+  - "https://www.bleepingcomputer.com/news/security/hackers-abuse-lookalike-domains-and-favicons-for-credit-card-theft/"
+type: "rule"
+authors:
+  - twitter: "ajpc500"
+source: |
+  type.inbound
+  and any(body.links,  
+    regex_search(.href_url.domain.domain, '\Axn--', '[^\x00-\x7F]')
+  ) 

```

Comment:
```suggestion
  ) 
  and (
    // include automated emails
    sender.display_name == "WordPress"
    or sender.email.local_part == "wordpress"

    // Has my organization ever sent an email to this sender?
    or (
        sender.email.domain.root_domain in $free_email_providers
        and sender.email.email not in $recipient_emails
    )
    or (
        sender.email.domain.root_domain not in $free_email_providers
        and sender.email.domain.domain not in $recipient_domains
    )
  )
```

Reply:
Yep, happy with that! We've seen a few too, largely from other regions e.g. Japan. This would filter those and still hit the TPs we've seen 👍🏻

Reply:
awesome 🤘